src/manifold.cpp

```diff
@@ -415,6 +415,29 @@ Manifold Manifold::SetTolerance(double tolerance) const {
   return Manifold(impl);
 }
 
+/**
+ * Return a copy of the manifold simplified to the given tolerance, but with its
+ * actual tolerance value unchanged. If no tolerance is given, the current
+ * tolerance is used for simplification.
+ */
+Manifold Manifold::Simplify(double tolerance) const {
+  auto impl = std::make_shared<Impl>(*GetCsgLeafNode().GetImpl());
+  const double oldTolerance = impl->tolerance_;
+  if (tolerance == 0) tolerance = oldTolerance;
+  if (tolerance > oldTolerance) {
+    impl->tolerance_ = tolerance;
+    impl->CreateFaces();
+    impl->SimplifyTopology();
+    impl->Finish();
+  }
+  impl->tolerance_ = oldTolerance;
+  return Manifold(impl);
+  // const double oldTolerance = GetCsgLeafNode().GetImpl()->tolerance_;
+  // Manifold result = SetTolerance(tolerance);
+  // result.GetCsgLeafNode().GetImpl()->tolerance_ = oldTolerance;
```

Comment:
@pca006132 What do you think is the right pattern to reduce duplication between these functions? The one I've commented out here didn't compile.

Reply:
eh, I guess this is fine - only a few lines duplicated really.